frappe/model/document.py

```diff
@@ -1492,12 +1493,14 @@ def lock(self, timeout=None):
 				raise frappe.DocumentLockedError
 		file_lock.create_lock(signature)
 		frappe.local.locked_documents.append(self)
+		self.is_locked = True
 
 	def unlock(self):
 		"""Delete the lock file for this document"""
 		file_lock.delete_lock(self.get_signature())
 		if self in frappe.local.locked_documents:
 			frappe.local.locked_documents.remove(self)
+			self.is_locked = False
 
```

Comment:
revert?

Reply:
I feel like a doc property should exist telling the status of the job? 🤔